gtsam/slam/ProjectionFactor.h

```diff
@@ -159,15 +159,20 @@ namespace gtsam {
     }
 
     /** return the measurement */
-    const Point2& measured() const {
+    auto measured() const -> const Point2& {
       return measured_;
     }
 
     /** return the calibration object */
-    inline const boost::shared_ptr<CALIBRATION> calibration() const {
+    auto calibration() const -> const boost::shared_ptr<CALIBRATION>& {
       return K_;
     }
 
+    /** return the (optional) sensor pose with respect to the vehicle frame */
+    auto body_P_sensor() const -> const boost::optional<POSE>& {
```

Comment:
While I think the trailing return type is great when using really complex return types (e.g. `Eigen::Block::ColXpr` which I personally used it for), the return types here are pretty straightforward and not verbose enough to merit the use of trailing return types.

Readability over convenience, I guess?

Reply:
Done in #842.